src/libexpr/eval.cc

```diff
@@ -896,6 +896,11 @@ void Value::mkPath(const SourcePath & path)
     mkPath(&*path.accessor, makeImmutableString(path.path.abs()));
 }
 
+void EvalState::registerAccessor(const ref<InputAccessor> accessor)
+{
+    inputAccessors.push_back(accessor);
+}
```

Comment:
This is like lazy-trees, except we only do it so that we don't destroy them when we put a non-smart pointer in `Value`, which has no finalizer because of GC and performance.